### PR TITLE
Rename ZIO.aggregateWithin to ZIO.aggregateWithinEither and add a simpler aggregateWithin

### LINKED ISSUE #1567: Rename ZIO.aggregateWithin to ZIO.aggregateWithinEither and add a simpler aggregateWithin
The current version of `aggregateWithin` emits the schedule's output into the stream, resulting in a stream of `Either[ScheduleOutput, Batch]`. This variant should be named `aggrgateWithinEither`, and `ZStream#aggregateWithin` should be defined as:
```scala
def aggregateWithin = aggregateWithinEither.collect {
  case Right(v) => v
}
```

### PR DESCRIPTION
**Pull request by saeltz**
This fixes #1567

### CODE REVIEW

**Comment by iravid** [comment]
Needs a rebase on master

**Comment by saeltz** [comment]
Finally, ready for your review, @iravid.

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -250,7 +250,7 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
    * Aggregated elements will be fed into the schedule to determine the delays between
    * pulls.
    */
-  final def aggregateWithin[R1 <: R, E1 >: E, A1 >: A, B, C](
+  final def aggregateWithinEither[R1 <: R, E1 >: E, A1 >: A, B, C](
```
Comment (vasilmkd) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: We will need to add some more documentation to this method, to highlight the difference compared to the new `aggregateWithin` which currently doesn't have a scaladoc.

Comment (saeltz) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: I appreciate your feedback! I've added ScalaDoc for both methods in https://github.com/zio/zio/pull/1700/commits/a4b8851cafcfc94722a5f88180fec3d56bad1abe.

Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: @iravid Should we open an issue maybe for the hackathon, to add annotated documentation like this?

Comment (saeltz) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: See https://github.com/zio/zio/issues/413 where I started adding ScalaDoc to some files. It's a good first issue for a hackathon.

**Pull request merged**